jbmc/src/java_bytecode/expr2java.cpp

```diff
@@ -191,41 +191,33 @@ std::string expr2javat::convert_constant(
     std::string dest;
     dest.reserve(char_representation_length);
 
-    mp_integer int_value;
-    if(to_integer(src, int_value))
-      UNREACHABLE;
+    const wchar_t int_value = numeric_cast_v<wchar_t>(src);
 
```

Comment:
This is difficult. Java chars are fixed to be 16 bits, by the standard.
wchar_t may be 16 bits, or may be 32 bits, but there isn't an obvious connection to Java's concept of a 'char'.

Reply:
Also, signedness may differ. Java's char is unsigned.

Reply:
Hmm, thanks, I had inferred the type from what `utf16_native_endian_to_java` takes as parameter. What would be the right type to use here, and where should a possible cast happen?

Reply:
Anything that is guaranteed to hold an unsigned 16-bit values should work -- so how about "char16_t". I would then do an explicit cast to wchar_t when doing the call to utf16_native_endian_to_java. Really utf16_native_endian_to_java should take a char16_t.

Reply:
Yes `char16_t` should be used here and in `utf16_native_endian_to_java`.